src/dragonball/src/vm/kernel_config.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2022 Alibaba Cloud. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::fs::File;
+
+/// Structure to hold guest kernel configuration information.
+pub struct KernelConfigInfo {
+    /// The descriptor to the kernel file.
+    kernel_file: File,
+    /// The descriptor to the initrd file, if there is one
+    initrd_file: Option<File>,
+    /// The commandline for guest kernel.
+    cmdline: linux_loader::cmdline::Cmdline,
+    /// Share guest kernel text/ro sections.
+    share_ro_sections: String,
```

Comment:
Why is this a `String` and not just a `bool`?

Reply:
This is a performance feature and we decide to not include this in the stage one for Dragonball:)